custom/ewsghana/reports/email_reports.py

```diff
@@ -0,0 +1,222 @@
+from casexml.apps.stock.models import StockTransaction
+from corehq.apps.commtrack.models import StockState, CommtrackConfig
+from corehq.apps.locations.models import SQLLocation
+from corehq.apps.reports.commtrack.const import STOCK_SECTION_TYPE
+from corehq.apps.reports.datatables import DataTablesHeader, DataTablesColumn
+from corehq.apps.reports.filters.fixtures import AsyncLocationFilter
+from corehq.apps.reports.filters.dates import DatespanFilter
+from custom.ewsghana.reports import EWSData, MultiReport
+from django.utils.translation import ugettext as _
+from custom.ewsghana.utils import get_supply_points
+from dimagi.utils.decorators.memoized import memoized
+
+
+def get_stock_summary_locations(loc_id, domain):
+    return get_supply_points(loc_id, domain)
+
+
+def get_rms_cms_summary_locations(domain):
+    return SQLLocation.objects.filter(location_type__in=['Regional Medical Store', 'Central Medical Store'],
+                                      domain=domain)
+
+
+class EmailReportData(EWSData):
+    show_table = True
+    use_datatables = True
+
+    def get_locations(self, loc_id, domain):
+        return []
+
+    @property
+    def headers(self):
+        return DataTablesHeader(*[
+            DataTablesColumn(_('Product')),
+            DataTablesColumn(_('% Facilities with Stockouts')),
+            DataTablesColumn(_('Total # Facilities Registered with this Product')),
+            DataTablesColumn(_('Total Stock')),
+            DataTablesColumn(_('% Facilities with Consumption Data')),
+            DataTablesColumn(_('Monthly Consumption')),
+            DataTablesColumn(_('Months of Stock')),
+            DataTablesColumn(_('Stock Status'))
+        ])
+
+    @property
+    def rows(self):
+        def percent(x, y):
+            return "%d%% (%d)" % (x * 100 / (y or 1), x)
+
+        def stock_status(status):
+            stock_levels = CommtrackConfig.for_domain(self.config['domain']).stock_levels_config
+            if status == 0.0:
+                return 'Stockout'
```

Comment:
this whole PR has a pretty inconsistent use of translations. doesn't matter since the reports aren't going to be in another language, but just noting